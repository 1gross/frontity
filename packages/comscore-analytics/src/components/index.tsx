import React from "react";
import { Head, connect } from "frontity";
import { Connect } from "frontity/types";
import ComscoreAnalytics from "../../types";

const ComscoreHead: React.FC<{ id?: string }> = ({ id }) => (
  <Head>
    <noscript>
      {`<img alt="comscore" src="https://sb.scorecardresearch.com/p?c1=2&c2=${id}&cv=2.0&cj=1" />`}
    </noscript>
    <script async src="https://sb.scorecardresearch.com/beacon.js" />
  </Head>
);

export const Root: React.FC<Connect<ComscoreAnalytics>> = ({ state }) => {
  return (
    <>
      {state.comscoreAnalytics.trackingIds.map((id) => (
        <ComscoreHead id={id} key={id} />
      ))}
    </>
  );
};

export default connect(Root);
