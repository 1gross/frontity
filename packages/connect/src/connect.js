import { createContext, useContext, createElement } from "react";
import { view } from "@risingstack/react-easy-state";

/**
 * A React context that stores the store of Frontity Connect.
 */
const context = createContext();

/**
 * The Provider of the store context, to be used in the root of the React
 * application.
 */
export const Provider = context.Provider;

/**
 * Connects a React component with the Frontity connect store.
 *
 * @param Component The React component to be connected
 * @returns The same Component, but with the store included in the props.
 */
export const connect = (Component) => {
  const isStatelessComp = !(
    Component.prototype && Component.prototype.isReactComponent
  );

  if (isStatelessComp) {
    return view((props) => {
      const store = useContext(context);
      return Component({ ...props, ...store });
    });
  } else {
    class ConnectedComponent extends Component {
      static contextType = context;

      render() {
        return createElement(Component, {
          ...this.props,
          ...this.context,
        });
      }
    }

    return view(ConnectedComponent);
  }
};
