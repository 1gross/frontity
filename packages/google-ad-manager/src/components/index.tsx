import { Head } from "frontity";
import React from "react";
import GoogleAdManager from "../../types";

/**
 * Documented at {@link GoogleAdManager}.
 *
 * @returns React element.
 */
const Root: GoogleAdManager["roots"]["googleAdManager"] = () => {
  return (
    <Head>
      <script
        src="https://securepubads.g.doubleclick.net/tag/js/gpt.js"
        async
      />
    </Head>
  );
};

export default Root;
