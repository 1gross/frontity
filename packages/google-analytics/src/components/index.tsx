/* eslint-disable react/no-danger */
import React from "react";
import { Head, connect } from "frontity";
import { Connect } from "frontity/types";
import GoogleAnalytics from "../../types";
import { getTrackerName } from "..";

const Root: React.FC<Connect<GoogleAnalytics>> = ({ state }) => {
  // Get Tracking ids from state.
  const { trackingIds, trackingId } = state.googleAnalytics;
  const ids = trackingIds || (trackingId && [trackingId]) || [];

  // Render Tracker code.
  return (
    <Head>
      <script>{`
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
${ids.map(
  id => `ga('create', '${id}', 'auto', '${getTrackerName(id)}');
`
)}`}</script>
    </Head>
  );
};

export default connect(Root);
