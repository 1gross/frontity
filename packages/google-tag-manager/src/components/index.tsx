/* eslint-disable react/no-danger */
import React from "react";
import { Head, connect } from "frontity";
import { Connect } from "frontity/types";
import GoogleTagManager from "../../types";

const GtmScript = ({ containerId }: { containerId: string }) => (
  <Head>
    <script
      async
      src={`https://www.googletagmanager.com/gtm.js?id=${containerId}`}
    />
  </Head>
);

const GtmNoScript = ({ containerId }: { containerId: string }) => (
  <Head>
    <noscript>
      {`
  <iframe
    src="https://www.googletagmanager.com/ns.html?id=${containerId}"
    height="0"
    width="0"
    style="display: 'none', visibility: 'hidden'"
  ></iframe>`}
    </noscript>
  </Head>
);

export const Root: React.FC<Connect<GoogleTagManager>> = ({ state }) => {
  const { containerId, containerIds } = state.googleTagManager;
  const ids = containerIds || (containerId && [containerId]) || [];

  return (
    <>
      {ids.map((id) => (
        <React.Fragment key={id}>
          <GtmScript containerId={id} />
          <GtmNoScript containerId={id} />
        </React.Fragment>
      ))}
    </>
  );
};

export default connect(Root);
