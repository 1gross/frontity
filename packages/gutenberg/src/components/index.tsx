import React from "react";
import { connect } from "frontity";

const Gutenberg = () => {
  return <>{/* <Style />
      <Theme /> */}</>;
};

export default connect(Gutenberg);
