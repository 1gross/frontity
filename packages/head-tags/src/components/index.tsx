import React from "react";
import { connect, Head } from "frontity";
import { Connect } from "frontity/types";
import HeadTagsPackage from "../../types";

// Render all head tags from the current entity.
const Root: React.FC<Connect<HeadTagsPackage>> = ({ state }) => {
  return (
    <Head>
      {state.headTags
        .get(state.router.link)
        .map(({ tag: Tag, attributes, content }, index) => {
          return (
            <Tag key={index} {...attributes}>
              {content}
            </Tag>
          );
        })}
    </Head>
  );
};

export default connect(Root);
