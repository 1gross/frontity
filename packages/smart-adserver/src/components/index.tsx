import React from "react";
import { Head, connect } from "frontity";

/**
 * Root of the Smartads package.
 *
 * @param props - Frontity injected props.
 * @returns React element.
 */
const SmartAdserver = ({ state }) => {
  const { networkId, subdomain } = state.smartAdserver;

  React.useEffect(() => {
    window.sas = window.sas || { cmd: [] };
    window.sas.cmd.push(function () {
      window.sas.setup({
        networkid: networkId,
        domain: `//${subdomain}.smartadserver.com`,
        async: true,
      });
    });
  }, []);

  return (
    <Head>
      <script src={`//ced.sascdn.com/tag/${networkId}/smart.js`} async />
    </Head>
  );
};

export default connect(SmartAdserver);
