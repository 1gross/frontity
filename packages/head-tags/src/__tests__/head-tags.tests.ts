import clone from "clone-deep";
import { createStore, InitializedStore } from "@frontity/connect";
import wpSource from "@frontity/wp-source/src";
import HeadTagsPackage, { PostTypeWithHeadTags } from "../../types";
import headTagsPackage from "..";
import { getCurrentHeadTags } from "../components";

// Import all mocks.
import postType from "./mocks/post-type.json";

let store: InitializedStore<HeadTagsPackage>;
beforeEach(() => {
  // Create store.
  const config: HeadTagsPackage = clone(headTagsPackage());

  // Mock wp-source state.
  const {
    state: { source }
  } = clone(wpSource());
  config.state.source = source;
  config.state.source.api = "https://test.frontity.io/wp-json";

  // Mock router state.
  config.state.router = { link: "/" };

  // Mock site url.
  config.state.frontity = { url: "https://mars.frontity.org" };

  // Initialize store.
  store = createStore(config);
});

describe("HeadTags component", () => {
  test("works with post archive", () => {
    // Populate source state.
    store.state.source.type = {
      post: postType as PostTypeWithHeadTags
    };
    store.state.source.data = {
      "/": {
        type: "post",
        items: [],
        isArchive: true,
        isPostTypeArchive: true,
        isPostArchive: true,
        isHome: true,
        isFetching: false,
        isReady: true
      }
    };

    // Populate router state.
    store.state.router.link = "/";

    // Test current head tags.
    expect(getCurrentHeadTags({ ...store })).toMatchSnapshot();
  });
  test.todo("add tests");
});
