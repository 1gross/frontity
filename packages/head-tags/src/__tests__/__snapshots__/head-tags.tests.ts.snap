// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadTags component works with post archive 1`] = `
Array [
  Object {
    "content": "frontity - Just another WordPress site",
    "tag": "title",
  },
  Object {
    "attributes": Object {
      "content": "Just another WordPress site",
      "name": "description",
    },
    "tag": "meta",
  },
  Object {
    "attributes": Object {
      "content": "max-snippet:-1, max-image-preview:large, max-video-preview:-1",
      "name": "robots",
    },
    "tag": "meta",
  },
  Object {
    "attributes": Object {
      "href": "https://mars.frontity.org",
      "rel": "canonical",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "href": "https://mars.frontity.org/page/2/",
      "rel": "next",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "content": "en_US",
      "property": "og:locale",
    },
    "tag": "meta",
  },
  Object {
    "attributes": Object {
      "content": "website",
      "property": "og:type",
    },
    "tag": "meta",
  },
  Object {
    "attributes": Object {
      "content": "frontity - Just another WordPress site",
      "property": "og:title",
    },
    "tag": "meta",
  },
  Object {
    "attributes": Object {
      "content": "Just another WordPress site",
      "property": "og:description",
    },
    "tag": "meta",
  },
  Object {
    "attributes": Object {
      "content": "https://mars.frontity.org",
      "property": "og:url",
    },
    "tag": "meta",
  },
  Object {
    "attributes": Object {
      "content": "frontity",
      "property": "og:site_name",
    },
    "tag": "meta",
  },
  Object {
    "attributes": Object {
      "content": "summary_large_image",
      "name": "twitter:card",
    },
    "tag": "meta",
  },
  Object {
    "attributes": Object {
      "content": "Just another WordPress site",
      "name": "twitter:description",
    },
    "tag": "meta",
  },
  Object {
    "attributes": Object {
      "content": "frontity - Just another WordPress site",
      "name": "twitter:title",
    },
    "tag": "meta",
  },
  Object {
    "attributes": Object {
      "class": "yoast-schema-graph yoast-schema-graph--main",
      "type": "application/ld+json",
    },
    "content": "{\\"@context\\":\\"https://schema.org\\",\\"@graph\\":[{\\"@type\\":\\"WebSite\\",\\"@id\\":\\"https://mars.frontity.org#website\\",\\"url\\":\\"https://mars.frontity.org\\",\\"name\\":\\"frontity\\",\\"potentialAction\\":{\\"@type\\":\\"SearchAction\\",\\"target\\":\\"https://mars.frontity.org?s={search_term_string}\\",\\"query-input\\":\\"required name=search_term_string\\"}},{\\"@type\\":\\"CollectionPage\\",\\"@id\\":\\"https://mars.frontity.org#webpage\\",\\"url\\":\\"https://mars.frontity.org\\",\\"inLanguage\\":\\"en-US\\",\\"name\\":\\"frontity - Just another WordPress site\\",\\"isPartOf\\":{\\"@id\\":\\"https://mars.frontity.org#website\\"},\\"description\\":\\"Just another WordPress site\\"}]}",
    "tag": "script",
  },
  Object {
    "attributes": Object {
      "href": "//s.w.org",
      "rel": "dns-prefetch",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "href": "https://test.frontity.io/feed/",
      "rel": "alternate",
      "title": "frontity » Feed",
      "type": "application/rss+xml",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "href": "https://test.frontity.io/comments/feed/",
      "rel": "alternate",
      "title": "frontity » Comments Feed",
      "type": "application/rss+xml",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "href": "https://test.frontity.io/wp-json/",
      "rel": "https://api.w.org/",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "href": "https://test.frontity.io/xmlrpc.php?rsd",
      "rel": "EditURI",
      "title": "RSD",
      "type": "application/rsd+xml",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "href": "https://test.frontity.io/wp-includes/wlwmanifest.xml",
      "rel": "wlwmanifest",
      "type": "application/wlwmanifest+xml",
    },
    "tag": "link",
  },
  Object {
    "attributes": Object {
      "content": "WordPress 5.3",
      "name": "generator",
    },
    "tag": "meta",
  },
]
`;
